src/main/java/com/github/manolo8/darkbot/core/api/DarkBoatAdapter.java

```diff
@@ -24,6 +22,15 @@ public void createWindow() {
         apiThread.start();
     }
 
+    @Override
+    public void relogin() {
+        super.relogin();
+        String url = "https://" + loginData.getUrl() + "/",
+                sid = "dosid=" + loginData.getSid();
+
+        API.setData(url, sid, loginData.getPreloaderUrl(), loginData.getParams());
```

Comment:
make a setData function with this, and reuse it in createWindow and here (consider the super's relogin utilizing that directly